### PR TITLE
Rounding functionality

### PR DESCRIPTION
**Pull request by EricRBedard**
Adding option to round returned values (code to do this is in the Controller)

Many files affected to change the aggregation method return value from String to List<Double>

Tests are completed

### CODE REVIEW

File: src/main/java/metrics/report_format/JSON.java [hunk 1]
```diff
@@ -44,9 +44,19 @@ public String format(CompareDTO dto, List<Double> similarities, List<String> agg
...
+            StringBuilder data = new StringBuilder();
+            data.append("[ ");
+            for(int i = 0; i < aggregationValues.size(); i++){
+                if(i != 0){
+                    data.append(", ");
+                }
+                data.append(aggregationValues.get(i));
+            }
+            data.append(" ]");
+            aggregationsJson.addProperty(aggregationKey.toLowerCase(), data.toString());
```
Comment (CameronRushton) [inline src/main/java/metrics/report_format/JSON.java]: I'd recommend replacing this code with
```java
aggregationsJson.addProperty(aggregationKey.toLowerCase(), aggregationPairs.get(aggregationKey).toString());
```
And updating the two(?) tests that check these values to not have spaces in the array.

File: src/test/java/core/ControllerCompareTest.java [hunk 2]
```diff
@@ -628,12 +634,12 @@ public void testMultipleReportFormatsPrintResults() throws IOException {
...
 
         assertTrue(jsonResult.isJsonObject());
-        assertEquals("4.0", jsonResult.getAsJsonObject().get("results").getAsJsonObject().get("averagevalue").getAsString());
+        assertEquals("[ 4.0 ]", jsonResult.getAsJsonObject().get("results").getAsJsonObject().get("averagevalue").getAsString());
 
         deleteFiles(RESULT_FILE_NAME + "RawResults", RESULT_FILE_NAME + "JSON");
 
         // Check console
-        assertTrue("Correct result wasn't printed to console.", outContent.toString().contains("4.0"));
+        assertTrue("Correct result wasn't printed to console.", outContent.toString().contains("[ 4.0 ]"));
```
Comment (CameronRushton) [inline src/test/java/core/ControllerCompareTest.java]: These are two of the places where I was saying in another comment these spaces in the array aren't necessary.

File: src/test/java/metrics/report_format/JSONTest.java [hunk 3]
```diff
@@ -59,9 +62,9 @@ public void testCreateJson() {
...
         assertTrue("No 'results' parameter found.", json.getAsJsonObject().has("results"));
         assertTrue("No 'aggregationmethod1' parameter found in results.", json.getAsJsonObject().getAsJsonObject("results").has("aggregationmethod1"));
-        assertEquals("No first aggregated value found in results.", "aggregatedvalue1", json.getAsJsonObject().getAsJsonObject("results").get("aggregationmethod1").getAsString());
+        assertEquals("No first aggregated value found in results.", "[ " + AGGREGATED_VALUE_ONE + " ]", json.getAsJsonObject().getAsJsonObject("results").get("aggregationmethod1").getAsString());
         assertTrue("No 'aggregationmethod2' parameter found in results.", json.getAsJsonObject().getAsJsonObject("results").has("aggregationmethod2"));
-        assertEquals("No second aggregated value found in results.", "aggregatedvalue2", json.getAsJsonObject().getAsJsonObject("results").get("aggregationmethod2").getAsString());
+        assertEquals("No second aggregated value found in results.", "[ " + AGGREGATED_VALUE_TWO + " ]", json.getAsJsonObject().getAsJsonObject("results").get("aggregationmethod2").getAsString());
```
Comment (CameronRushton) [inline src/test/java/metrics/report_format/JSONTest.java]: This is the other place with array spaces.

File: src/main/java/metrics/report_format/JSON.java (hunk 1 above)
Comment (EricRBedard) [reply src/main/java/metrics/report_format/JSON.java]: makes it look a lot cleaner, thanks

**Pull request merged**